crates/lang/src/bql.pest

```diff
@@ -119,7 +119,52 @@ as_qualif = {
     name
 }
 
-from = { ^"from" ~ ( join_tables | compound_select ) }
+host_subname = { ASCII_ALPHANUMERIC ~ ( ( ASCII_ALPHANUMERIC ~ HYPHEN ~ ASCII_ALPHANUMERIC ) | ASCII_ALPHANUMERIC )* }
+domain = { ASCII_ALPHA_LOWER+ }
+host_name = { (host_subname ~ ".")* ~ domain }
+
+ipv4_lit = @{ ASCII_DIGIT{1,3} ~ ( "." ~ ASCII_DIGIT{1,3} ){3} }
+ipv6_lit = @{
+    ( ":" | ASCII_ALPHANUMERIC{1,4} ) ~ ":" ~ ( ipv4_lit | ASCII_ALPHANUMERIC{1,4} | ":" )*
+}
+
+from = { ^"from" ~ ( remote_func | join_tables | compound_select ) }
+
+remote_func = {
+    ^"remote" ~ "(" ~ remote_addresses ~ "," ~ remote_table ~ ("," ~ username_lit ~ "," ~ password_lit)? ~ ")"
+}
+
+remote_table = ${
+  (remote_database_name ~ ".")? ~ remote_table_name
+}
+
+remote_database_name = @{ name }
+remote_table_name = @{ name }
+username = @{ ASCII_ALPHANUMERIC* }
+password = @{ ASCII_ALPHANUMERIC* }
+
+username_lit = {"'" ~ username ~ "'"}
+password_lit = {"'" ~ password ~ "'"}
+
+ip = {
+   ipv4_lit | ("[" ~ ipv6_lit ~ "]")
+}
+
+remote_addresses = {
+   "'" ~ remote_address ~ ("," ~ remote_address)* ~ "'"
+ }
+
+ip_address = { ip ~ (":" ~ port)? }
+host_address = { host_name ~ (":" ~ port)? }
+
+remote_address = {
+  ip_address | host_address
+}
+
+port = @{
+    ASCII_DIGIT{1,5}
+}
+
```

Comment:
These new additions seem ok. 

Generally, for parsing, one rule is that we do not do too much complex verifications. Because we will verify it in the later by using them (access/connection/...). Too much complex parsing rules are time-cost. But for most cases, people do right thing.